src/dmd/backend/barray.d

```diff
@@ -0,0 +1,85 @@
+/**
+ * Compiler implementation of the
+ * $(LINK2 http://www.dlang.org, D programming language).
+ *
+ * Copyright:   Copyright (C) 2018 by The D Language Foundation, All Rights Reserved
+ * Authors:     $(LINK2 http://www.digitalmars.com, Walter Bright)
+ * License:     $(LINK2 http://www.boost.org/LICENSE_1_0.txt, Boost License 1.0)
+ * Source:      $(LINK2 https://github.com/dlang/dmd/blob/master/src/dmd/backend/barrayf.d, backend/barray.d)
+ * Documentation: https://dlang.org/phobos/dmd_backend_barray.html
+ */
+
+module dmd.backend.barray;
+
+import core.stdc.stdio;
+import core.stdc.stdlib;
+import core.stdc.string;
+
+nothrow:
+
+extern (C++): void err_nomem();
+
+/*************************************
+ * A reusable array that ratchets up in capacity.
+ */
+struct Barray(T)
```

Comment:
I'm guessing `Barray` for backend array? I don't like that name, it defeats the purpose of modules.